src/algorithmForNN/pointCloudUtils.py

```diff
@@ -151,6 +151,104 @@ def get_area_of_plane(points: o3d.geometry.PointCloud, plane_model: list) -> flo
     return current_list[1] * current_list[2]
 
 
+def get_calibration_matrix_from_calib_file(path_to_calibration_file: str) -> np.ndarray:
+    calib_matrix = np.zeros((4, 4))
+    with open(path_to_calibration_file) as file:
+        (
+            calib_matrix[0, 0],
+            calib_matrix[0, 1],
+            calib_matrix[0, 2],
+            calib_matrix[0, 3],
+            calib_matrix[1, 0],
+            calib_matrix[1, 1],
+            calib_matrix[1, 2],
+            calib_matrix[1, 3],
+            calib_matrix[2, 0],
+            calib_matrix[2, 1],
+            calib_matrix[2, 2],
+            calib_matrix[2, 3],
+        ) = list(map(float, file.readlines()[4][4:].rstrip().split(" ")))
+        calib_matrix[3, 3] = 1
+
+    return calib_matrix
+
+
+def get_position_from_poses_file(
+    path_to_poses_file: str, frame_number: int
+) -> np.ndarray:
+    pose_matrix = np.zeros((4, 4))
+    with open(path_to_poses_file) as file:
+        lines = file.readlines()
+        (
+            pose_matrix[0, 0],
+            pose_matrix[0, 1],
+            pose_matrix[0, 2],
+            pose_matrix[0, 3],
+            pose_matrix[1, 0],
+            pose_matrix[1, 1],
+            pose_matrix[1, 2],
+            pose_matrix[1, 3],
+            pose_matrix[2, 0],
+            pose_matrix[2, 1],
+            pose_matrix[2, 2],
+            pose_matrix[2, 3],
+        ) = list(map(np.float64, lines[frame_number].rstrip().split(" ")))
+        pose_matrix[3, 3] = 1
+
+    return pose_matrix
+
+
+def transform_positions_in_point_cloud(
+    calib_matrix: np.ndarray, pose_matrix: np.ndarray, point_claud: np.ndarray
+) -> np.ndarray:
+    left_camera_matrix = np.matmul(pose_matrix, calib_matrix)
+
+    for index, point in enumerate(point_claud):
+        temp_point = np.array([[point[0], point[1], point[2], 1]]).T
+        new_point = np.matmul(left_camera_matrix, temp_point)
+        point_claud[index, 0], point_claud[index, 1], point_claud[index, 2] = (
+            new_point[0, 0],
+            new_point[1, 0],
+            new_point[2, 0],
+        )
+
+    return point_claud
+
+
+def create_point_cloud_by_first_N_snapshots(
```

Comment:
i think it is not an util function

Reply:
where do you suggest to move this function?

Reply:
Some special file --- where algo of pcd combining will be